notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+
+Let's compare the performance of three different classifiers:
+* `ham_only`: predicts 'ham' for every email.
+* `spam_only`: predicts 'spam' for every email.
+* `words_list_model`: predicts 'ham' or 'spam' based on the presence of certain words in the body of an email. 
+
+We construct `words_list_model` using the following procedure: Suppose we have a list of words `words_list` that we believe are common in spam emails: "please", "click", "money", "business", and "remove". We transform each email into a feature vector by setting the vector's $i$th entry to 1 if the $i$th word in `words_list` is contained in the email body and 0 if it isn't. For example, using our five chosen words and the email body "please remove by tomorrow", the feature vector would be $[1, 0, 0, 0, 1]$. This procedure generates the `1000 X 5` feature matrix $X$.
+
+The following code block displays the accuracies of the classifiers. Model creation and training are omitted for brevity.
+
+
+```python
+# HIDDEN
+
+words_list = ['please', 'click', 'money', 'business', 'remove']
+
+X = pd.DataFrame(words_in_texts(words_list, emails['email'].str.lower())).as_matrix()
+y = emails['spam'].as_matrix()
+
+# Train-test split
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, random_state=41, test_size=0.2
+)
+
+#Fit the model
+words_list_model = LogisticRegression(fit_intercept=True)
+words_list_model.fit(X_train, y_train)
+
+y_prediction_words_list = words_list_model.predict(X_test)
+y_prediction_ham_only = np.zeros(len(y_test))
+y_prediction_spam_only = np.ones(len(y_test))
+```
+
+
+```python
+# This cell will be visible
+from sklearn.metrics import accuracy_score
+
+# Our selected words
+words_list = ['please', 'click', 'money', 'business']
+
+print(f'words_list_model test set accuracy: {np.round(accuracy_score(y_prediction_words_list, y_test), 3)}')
+print(f'ham_only test set accuracy: {np.round(accuracy_score(y_prediction_ham_only, y_test), 3)}')
+print(f'spam_only test set accuracy: {np.round(accuracy_score(y_prediction_spam_only, y_test), 3)}')
+```
+
+    words_list_model test set accuracy: 0.96
+    ham_only test set accuracy: 0.96
+    spam_only test set accuracy: 0.04
+
+
+Using `words_list_model` classifies 96% of the test set emails correctly. Although this accuracy appears high, `ham_only` achieves the same accuracy by uniformally labeling everything as ham. This is cause for concern because the data suggests we can do just as well without a spam filter at all.
+
+Model accuracy alone can be a misleading indicator of model performance, so we must investigate further. For each of our models, we can compare the number of erroneous classifications to the number of correct classifications by building a **confusion matrix**. A confusion matrix for a binary classifier is a two-by-two table that contains the model predictions on one axis and the actual labels on the other. 
```

Comment:
Let's add a hook at the beginning of this sentence to make the transition smoother:

> As the accuracies above show, model accuracy alone can be a misleading indicator of model performance. We can understand the model's predictions in greater depth using a **confusion matrix**. A confusion matrix for a binary classifier is a two-by-two heatmap that contains the model predictions on one axis and the actual labels on the other.